### DISCUSSION TITLE
Anyone compile/run this in JetBrains RustRanger?

### DISCUSSION THREAD

**Discussion by wbpaley** [rust-tutorials/learn-opengl#15] [Q&A]
Excited to get started--thanks for making these tutorials!

Rust newbie here, though I've been coding computer graphics since '76, C++ since it first came out, used the original GL on serial number 000049 SGI Iris, & Java mostly since then (so you can talk to me like a third-grader instead of a kindergartener...  ; )

Created the project in RustRover using their create from VCS process & the GitHub link. It all gets downloaded & managed in a project but automatically tries to compile and I see:
_error: failed to run custom build command for `fermium v0.1.4`_

Has anyone brought this into RustRover & solved this startup/config (I'm guessing) issue?

Thanks!

**Comment by Lokathor** [comment]
Ah, didn't realize that the `beryllium` dependency (and the `fermium` dependency that it brings in) is so old! It's been a while indeed.

I'll see if I can get the windowing parts updated to avoid build problems, hopefully I can get it fixed tomorrow.

**Comment by wbpaley** [comment]
That would be amazing; thanks!

I’m on the Mac Apple Silicon hardware platform if that’s of any use.





.

**Comment by Lokathor** [comment]
I've updated the code to use the latest version of beryllium

This isn't a sure-thing because honestly I've never experimented with an Apple M1 before, but it's more likely to work now.

**Comment by Lokathor** [comment]
Also let's make this an issue: https://github.com/rust-tutorials/learn-opengl/issues/16

**Comment by wbpaley** [comment]
Thanks! But little change.

Updated and it automatically tried to compile again with the same error plus the updated version number:
_error: failed to run custom build command for `fermium v22605.0.0`_

Just to be sure I did something my colleagues & I do in InteliJ: cleared all caches & restarted--same auto-compile & same error.

Screen grab if it's useful
![Screenshot 2023-12-05 at 9 55 25 PM](https://github.com/rust-tutorials/learn-opengl/assets/60867/c3f58f28-ff97-4699-87f9-e29e4fc99a2b)

It's a MacBook Pro with M2, not M1, but I think that's just more cores & the same AArch64 architecture...

Anyone out there able to try on any Apple Silicon?

**Comment by wbpaley** [comment]
Still greatly appreciate the quick response! But don't spend more time on this issue on my account.

I needed to switch to WebGPU at some point anyway, and was able to get this running in RustRover, so I can take it from there.

GitHub project I used is this if anyone needs the same:
https://github.com/gfx-rs/wgpu

![Screenshot 2023-12-06 at 9 33 12 PM](https://github.com/rust-tutorials/learn-opengl/assets/60867/23789dd0-8d33-4378-9136-888473adf19b)
